kytos/core/auth.py

```diff
@@ -44,6 +56,120 @@ def wrapper(*args, **kwargs):
+        max=int(os.environ.get("MONGO_AUTO_RETRY_WAIT_RANDOM_MAX", 1)),
+    ),
+    before_sleep=before_sleep,
+    retry=retry_if_exception_type((AutoReconnect,)),
+)
+class UserController:
+    """UserController"""
+
+    # pylint: disable=unnecessary-lambda
+    def __init__(self, get_mongo=lambda: Mongo()):
+        self.mongo = get_mongo()
+        self.db_client = self.mongo.client
+        self.db = self.db_client[self.mongo.db_name]
+
+    def bootstrap_indexes(self):
+        """Bootstrap all users related indexes."""
+        index_tuples = [
+            ("users", [("username", pymongo.ASCENDING)], {"unique": True}),
+        ]
+        for collection, keys, kwargs in index_tuples:
+            if self.mongo.bootstrap_index(collection, keys, **kwargs):
+                LOG.info(
+                    f"Created DB index {keys}, collection: {collection}"
+                )
+
+    def create_user(self, user_data: dict) -> InsertOneResult:
+        """Create user to database"""
+        try:
+            utc_now = datetime.datetime.utcnow()
+            result = self.db.users.insert_one(UserDoc(**{
+                **user_data,
+                **{"inserted_at": utc_now},
+                **{"updated_at": utc_now}
+            }).dict())
+        except DuplicateKeyError as err:
+            raise err
+        except ValidationError as err:
+            raise err
+        return result
+
+    def delete_user(self, username: str) -> dict:
+        """Delete user from database"""
+        utc_now = datetime.datetime.utcnow()
+        result = self.db.users.find_one_and_update(
+            {"username": username},
+            {"$set": {
+                "state": "inactive",
+                "deleted_at": utc_now,
+            }},
+            return_document=ReturnDocument.AFTER,
+        )
+        return result
+
+    def update_user(self, username: str, data: dict) -> dict:
+        """Update user from database"""
+        utc_now = datetime.datetime.utcnow()
+        try:
+            result = self.db.users.find_one_and_update(
+                {"username": username},
+                {
+                    "$set": UserDocUpdate(**{
+                        **data,
+                        **{"updated_at": utc_now}
+                    }).dict(exclude_none=True)
+                },
+                return_document=ReturnDocument.AFTER
+            )
+        except ValidationError as err:
+            raise err
+        except DuplicateKeyError as err:
+            raise err
+        return result
+
+    def get_user(self, username: str) -> dict:
+        """Return a user information from database"""
+        data = self.db.users.aggregate([
+            {"$match": {"username": username}},
+            {"$project": UserDoc.projection()},
+            {"$limit": 1}
+        ])
+        try:
+            user, *_ = list(value for value in data)
+            return user
+        except ValueError:
+            return None
```

Comment:
The return type of `dict` and you're returning `None`, either we could return an empty dict, or update the return type to be `Optional[dict]`. Same for `get_user_nopw` below.

Reply:
Empty `dict` is better I think.